bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -1231,7 +1232,8 @@ public static boolean format(ServerConfiguration conf,
             boolean isInteractive, boolean force) throws Exception {
         return runFunctionWithMetadataBookieDriver(conf, driver -> {
             try {
-                boolean ledgerRootExists = driver.getRegistrationManager().prepareFormat();
+                RegistrationManager regManager = driver.createRegistrationManager();
```

Comment:
do we have to close this ?

Reply:
Yes good catch. I have done an audit of all calls to createRegistrationManager() and ensured that the lifetime of the registration manager is correctly handled.